x-pack/plugins/infra/public/components/asset_details/tabs/processes/processes.tsx

```diff
@@ -69,22 +70,28 @@ export const Processes = () => {
 
   const debouncedSearchOnChange = useMemo(() => {
     return debounce<(queryText: string) => void>((queryText) => {
-      setUrlState({ processSearch: queryText });
       setSearchText(queryText);
     }, 500);
-  }, [setUrlState]);
+  }, []);
 
   const searchBarOnChange = useCallback(
-    ({ query, queryText }) => {
-      setSearchBarState(query);
-      debouncedSearchOnChange(queryText);
+    ({ query, queryText, error: queryError }) => {
+      if (queryError) {
+        setSearchQueryError(queryError);
+      } else {
+        setUrlState({ processSearch: queryText });
+        setSearchQueryError(null);
+        setSearchBarState(query);
+        debouncedSearchOnChange(queryText);
```

Comment:
So many setters now. 🤷

Reply:
Yeah, agree, does not look great... it seems like the only essential ones are `searchText` and `error`, everything else can be derived from them. Extracting URL state and query state updates into a separate `useEffect` that reacts on search text changes might help but will lead to another re-render. I'll try to restructure this a bit next time I touch this code, did not want to start any refactoring with this bug fix.